diff --git a/make/autoconf/flags.m4 b/make/autoconf/flags.m4
index 1dbcbe05ed6..9ee8990b118 100644
--- a/make/autoconf/flags.m4
+++ b/make/autoconf/flags.m4
@@ -136,7 +136,7 @@ AC_DEFUN([FLAGS_SETUP_MACOSX_VERSION],
     if test "x$OPENJDK_TARGET_CPU_ARCH" = xaarch64; then
       MACOSX_VERSION_MIN=11.00.00
     else
-      MACOSX_VERSION_MIN=10.12.0
+      MACOSX_VERSION_MIN=10.8.0
     fi
     MACOSX_VERSION_MIN_NODOTS=${MACOSX_VERSION_MIN//\./}

diff --git a/make/conf/jib-profiles.js b/make/conf/jib-profiles.js
index 4ee43abe774..a1492ea80f9 100644
--- a/make/conf/jib-profiles.js
+++ b/make/conf/jib-profiles.js
@@ -446,7 +446,7 @@ var getJibProfilesProfiles = function (input, common, data) {
             target_cpu: "x64",
             dependencies: ["devkit", "gtest", "pandoc"],
             configure_args: concat(common.configure_args_64bit, "--with-zlib=system",
-                "--with-macosx-version-max=10.12.00",
+                "--with-macosx-version-max=10.8.00",
                 "--enable-compatible-cds-alignment",
                 // Use system SetFile instead of the one in the devkit as the
                 // devkit one may not work on Catalina.
